src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -1512,6 +1512,25 @@ theorem shiftRight_add {w : Nat} (x : BitVec w) (n m : Nat) :
   ext i
   simp [Nat.add_assoc n m i]
 
+@[simp]
+theorem shiftLeft_ushiftRight {x : BitVec w} {n : Nat} :
+  x >>> n <<< n = x &&& BitVec.allOnes w <<< n := by
+  induction n generalizing x
+  case zero =>
+    ext; simp
+  case succ n ih =>
+    rw [BitVec.shiftLeft_add, Nat.add_comm, BitVec.shiftRight_add, ih,
+       Nat.add_comm, BitVec.shiftLeft_add, BitVec.shiftLeft_and_distrib]
+    ext i
+    have hi₁ (_: 0 < i.val) : 1 + (i.val - 1) = i := by
+      rw [Nat.add_comm, Nat.sub_add_cancel]
+      omega
+    by_cases hw : w = 0
+    · simp [hw]
+    by_cases hi₂ : i.val = 0
```

Comment:
I'm assuming this is the second case of the `by_cases` in 1528? If so there should be an additional center dot in front of this `by_cases` and then according indentation etc.